app/src/viewmodels/components/navigation/TopBarViewModel.tsx

```diff
@@ -0,0 +1,31 @@
+import { Colors } from "@/constants/Colors";
+import { useThemeColor } from "@/hooks/useThemeColor";
+
+/**
+ * The top bar ViewModel helps displays the top bar component.
+ * @param colorType : color type for the icons
+ * @returns : a view model for the navigation bars
+ */
+export function useTopBarViewModel({
+  leftIcon,
+  rightIcon,
+  colorType = "white",
+}: {
+  leftIcon?: string;
+  rightIcon?: string;
+  colorType?: keyof typeof Colors.light & keyof typeof Colors.dark;
+}) {
+  const color = useThemeColor({}, colorType);
+
```

Comment:
I feel like the same is not very clear, could you change it or add documentation ?